platform/pddf/platform-api-pddf-base/sonic_platform_pddf_base/pddf_current_sensor.py

```diff
@@ -0,0 +1,95 @@
+try:
+    from sonic_platform_base.sensor_base import CurrentSensorBase, SensorBase
+    from subprocess import getstatusoutput
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+class PddfCurrentSensor(CurrentSensorBase):
+    """PDDF generic Current Sensor class"""
+    pddf_obj = {}
+    plugin_data = {}
+
+    def __init__(self, index, pddf_data=None, pddf_plugin_data=None):
+        if not pddf_data or not pddf_plugin_data:
+            raise ValueError('PDDF JSON data error')
+
+        self.pddf_obj = pddf_data
+        self.plugin_data = pddf_plugin_data
+        self.sensor_index = index + 1
+
+        self.platform = self.pddf_obj.get_platform()
+
+        self.sensor_obj_name = f"CURRENT{self.sensor_index}"
+        if self.sensor_obj_name not in self.pddf_obj.data:
+            raise ValueError(f"Current sensor {self.sensor_obj_name} not found in PDDF data")
+
+        self.sensor_obj = self.pddf_obj.data[self.sensor_obj_name]
+        self.display_name = self.sensor_data.get('dev_attr', {}).get('display_name', self.sensor_obj_name)
```

Comment:
There is a typo error in line 27. The self.display_name = self.sensor_data.get should change to self.display_name = self.sensor_obj.get